src/main/java/org/jabref/gui/entryeditor/EntryEditor.java

```diff
@@ -355,11 +356,27 @@ private void setupToolBar() {
 
         // Add menu for fetching bibliographic information
         ContextMenu fetcherMenu = new ContextMenu();
-        for (EntryBasedFetcher fetcher : WebFetchers.getEntryBasedFetchers(preferencesService.getImportFormatPreferences(), preferencesService.getFilePreferences(), databaseContext, preferencesService.getDefaultEncoding())) {
+        for (EntryBasedFetcher fetcher : WebFetchers.getEntryBasedFetchers(preferencesService.getImportSettingsPreferences(), preferencesService.getImportFormatPreferences(), preferencesService.getFilePreferences(), databaseContext, preferencesService.getDefaultEncoding())) {
             MenuItem fetcherMenuItem = new MenuItem(fetcher.getName());
             fetcherMenuItem.setOnAction(event -> fetchAndMerge(fetcher));
             fetcherMenu.getItems().add(fetcherMenuItem);
         }
+        // Treat the PdfMergeMetadataImporter separately since if the use never accepted or disabled grobid, we ask to enable it
+        MenuItem pdfMergeMetadataImporterMenuItem = new MenuItem("PDFmergemetadata");
+        pdfMergeMetadataImporterMenuItem.setOnAction(event -> {
+            if (!preferencesService.getImportSettingsPreferences().isGrobidEnabled() && !preferencesService.getImportSettingsPreferences().isGrobidOptOut()) {
```

Comment:
I would extract this to a kind of helper method and class so you can reuse it in several places
e..g GrobidChoiceDialog -> showConfirmationDialog

Reply:
So you vote to have it whenever Grobid is used instead of once at startup, right?

Reply:
Yep. Only when it's needed for the first time.